src/runtime_src/core/edge/drm/zocl/zocl_drv.c

```diff
@@ -788,13 +789,17 @@ static int zocl_drm_platform_probe(struct platform_device *pdev)
 	/* Initial sysfs */
 	rwlock_init(&zdev->attr_rwlock);
 	ret = zocl_init_sysfs(drm->dev);
-	if (ret)
+	if (ret) {
+		mutex_destroy(&zdev->zdev_xclbin_lock);
```

Comment:
I think you need to clean the error label and error out sequence. For example, if zocl_xclbin_init fail, you will still call zocl_xclbin_fini() where might panic the system because zdev->zdev_xclbin is not valid. After clean up, you may be able to put mutex_destroy(xclbin_lock) into one of the error label.

Reply:
thanks, will get this fixed and retest failure cases today.

Reply:
fixed